src/app/rosetta/test-agent/agent.ml

```diff
@@ -202,19 +202,58 @@ let construction_api_payment_through_mempool ~logger ~rosetta_uri
     Offline.Payloads.req ~logger ~rosetta_uri ~network_response ~operations
       ~metadata:metadata_res.metadata
   in
-  let%bind parse_res =
+  let%bind payloads_parse_res =
     Offline.Parse.req ~logger ~rosetta_uri ~network_response
       ~transaction:
         (`Unsigned
           payloads_res.Construction_payloads_response.unsigned_transaction)
   in
-  if not ([%equal: Operation.t list] operations parse_res.operations) then (
+  if not ([%equal: Operation.t list] operations payloads_parse_res.operations)
+  then (
     [%log debug]
       ~metadata:
         [ ("expected", [%to_yojson: Operation.t list] operations)
-        ; ("actual", [%to_yojson: Operation.t list] parse_res.operations) ]
+        ; ( "actual"
+          , [%to_yojson: Operation.t list] payloads_parse_res.operations ) ]
       "Construction_parse : Expected $expected, after payloads+parse $actual" ;
     failwith "Operations are not equal before and after payloads+parse" ) ;
+  let%bind signature =
```

Comment:
could you remove the `%bind` and the `Deferred.return` ?

Reply:
I'm binding here with the Deferred.return to bind over the Result part (we're in the Deferred.Result monad)